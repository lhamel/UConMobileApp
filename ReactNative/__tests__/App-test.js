import { enableFetchMocks } from 'jest-fetch-mock';
enableFetchMocks();

import 'react-native';
import React from 'react';

import Schedule from '../components/Schedule';

import Event from '../components/Event';
import Header from '../components/Header';
import MyLargeButton from '../components/MyLargeButton';
import MySmallButton from '../components/MySmallButton';
import MyTextBox from '../components/MyTextBox';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('tests jest', () => {
  expect(true).toEqual(true);
});

/*
it('Schedule renders correctly', () => {
  const tree = renderer.create(
    <Schedule />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Event renders correctly', () => {
  const tree = renderer.create(
    <Event />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Header renders correctly', () => {
  const tree = renderer.create(
    <Header />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
*/

it('MyLargeButton renders correctly', () => {
  const tree = renderer.create(
    <MyLargeButton />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('MySmallButton renders correctly', () => {
  const tree = renderer.create(
    <MySmallButton />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('MyTextBox renders correctly', () => {
  const tree = renderer.create(
    <MyTextBox />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
