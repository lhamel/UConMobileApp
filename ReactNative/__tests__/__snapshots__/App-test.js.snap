// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyLargeButton renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 20,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "red",
        "borderRadius": 50,
        "opacity": 1,
        "padding": 12,
        "width": 200,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
        }
      }
    />
  </View>
</View>
`;

exports[`MySmallButton renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 20,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "dodgerblue",
        "borderRadius": 50,
        "opacity": 1,
        "padding": 12,
        "width": 100,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
        }
      }
    />
  </View>
</View>
`;

exports[`MyTextBox renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 50,
      "marginBottom": 20,
      "marginLeft": 10,
      "marginRight": 10,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "grey",
        "borderRadius": 50,
        "borderWidth": 2,
        "fontSize": 16,
        "padding": 10,
        "paddingLeft": 20,
        "width": 300,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
